upstream-sha256: 28d552ccc0e08a031b47639f7189911878cd0a69af552ab9e94ffaab63c15384
actual-sha256: 668c0bcfd0f85dd962b2f8ed0818d8738683bf876a40da12c9c90ee9014b8dc5
--- upstream.expected
+++ tolk.actual
@@ -2,59 +2,62 @@
    1 Ops.CONST           :            dtypes.weakint                           []                               0
    2 Ops.CAST            :            dtypes.int                               ['0']                            dtypes.int
    3 Ops.INDEX           :            dtypes.float                             [0, 2]                           None
-   4 Ops.PARAM           :            dtypes.float                             []                               ParamArg(1, dtypes.float, 16, device='CPU')
-   5 Ops.CONST           :            dtypes.weakint                           []                               4
-   6 Ops.CAST            :            dtypes.int                               ['4']                            dtypes.int
-   7 Ops.SHRINK          :            dtypes.float                             [4, 2, 6]                        None
-   8 Ops.LOAD            :            dtypes.float                             [7]                              None
-   9 Ops.INDEX           :            dtypes.float                             [8, 2]                           None
-  10 Ops.CONST           :            dtypes.weakint                           []                               1
-  11 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
-  12 Ops.INDEX           :            dtypes.float                             [8, 11]                          None
-  13 Ops.ADD             :            dtypes.float                             [9, 12]                          None
-  14 Ops.CONST           :            dtypes.weakint                           []                               2
-  15 Ops.CAST            :            dtypes.int                               ['2']                            dtypes.int
-  16 Ops.INDEX           :            dtypes.float                             [8, 15]                          None
-  17 Ops.ADD             :            dtypes.float                             [13, 16]                         None
-  18 Ops.CONST           :            dtypes.weakint                           []                               3
-  19 Ops.CAST            :            dtypes.int                               ['3']                            dtypes.int
-  20 Ops.INDEX           :            dtypes.float                             [8, 19]                          None
-  21 Ops.ADD             :            dtypes.float                             [17, 20]                         None
-  22 Ops.SHRINK          :            dtypes.float                             [4, 6, 6]                        None
-  23 Ops.LOAD            :            dtypes.float                             [22]                             None
-  24 Ops.INDEX           :            dtypes.float                             [23, 2]                          None
-  25 Ops.ADD             :            dtypes.float                             [21, 24]                         None
-  26 Ops.INDEX           :            dtypes.float                             [23, 11]                         None
-  27 Ops.ADD             :            dtypes.float                             [25, 26]                         None
-  28 Ops.INDEX           :            dtypes.float                             [23, 15]                         None
-  29 Ops.ADD             :            dtypes.float                             [27, 28]                         None
-  30 Ops.INDEX           :            dtypes.float                             [23, 19]                         None
-  31 Ops.ADD             :            dtypes.float                             [29, 30]                         None
-  32 Ops.CONST           :            dtypes.weakint                           []                               8
-  33 Ops.CAST            :            dtypes.int                               ['8']                            dtypes.int
-  34 Ops.SHRINK          :            dtypes.float                             [4, 33, 6]                       None
-  35 Ops.LOAD            :            dtypes.float                             [34]                             None
-  36 Ops.INDEX           :            dtypes.float                             [35, 2]                          None
-  37 Ops.ADD             :            dtypes.float                             [31, 36]                         None
-  38 Ops.INDEX           :            dtypes.float                             [35, 11]                         None
-  39 Ops.ADD             :            dtypes.float                             [37, 38]                         None
-  40 Ops.INDEX           :            dtypes.float                             [35, 15]                         None
-  41 Ops.ADD             :            dtypes.float                             [39, 40]                         None
-  42 Ops.INDEX           :            dtypes.float                             [35, 19]                         None
-  43 Ops.ADD             :            dtypes.float                             [41, 42]                         None
-  44 Ops.CONST           :            dtypes.weakint                           []                               12
-  45 Ops.CAST            :            dtypes.int                               ['12']                           dtypes.int
-  46 Ops.SHRINK          :            dtypes.float                             [4, 45, 6]                       None
-  47 Ops.LOAD            :            dtypes.float                             [46]                             None
-  48 Ops.INDEX           :            dtypes.float                             [47, 2]                          None
-  49 Ops.ADD             :            dtypes.float                             [43, 48]                         None
-  50 Ops.INDEX           :            dtypes.float                             [47, 11]                         None
-  51 Ops.ADD             :            dtypes.float                             [49, 50]                         None
-  52 Ops.INDEX           :            dtypes.float                             [47, 15]                         None
-  53 Ops.ADD             :            dtypes.float                             [51, 52]                         None
-  54 Ops.INDEX           :            dtypes.float                             [47, 19]                         None
-  55 Ops.ADD             :            dtypes.float                             [53, 54]                         None
-  56 Ops.SQRT            :            dtypes.float                             [55]                             None
-  57 Ops.NEG             :            dtypes.float                             [56]                             None
-  58 Ops.STORE           :            dtypes.void                              [3, 57]                          None
-  59 Ops.SINK            :            dtypes.void                              [58]                             KernelInfo(name='r_16', applied_opts=(Opt(op=OptOps.SPLIT, axis=0, arg=(0, AxisType.UNROLL)),), opts_to_apply=None, estimates=None, beam=0)
+   4 Ops.CONST           :            dtypes.weakfloat                         []                               0.0
+   5 Ops.CAST            :            dtypes.float                             ['0.0']                          dtypes.float
+   6 Ops.PARAM           :            dtypes.float                             []                               ParamArg(1, dtypes.float, 16, device='CPU')
+   7 Ops.CONST           :            dtypes.weakint                           []                               4
+   8 Ops.CAST            :            dtypes.int                               ['4']                            dtypes.int
+   9 Ops.SHRINK          :            dtypes.float                             [6, 2, 8]                        None
+  10 Ops.LOAD            :            dtypes.float                             [9]                              None
+  11 Ops.INDEX           :            dtypes.float                             [10, 2]                          None
+  12 Ops.ADD             :            dtypes.float                             [5, 11]                          None
+  13 Ops.CONST           :            dtypes.weakint                           []                               1
+  14 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
+  15 Ops.INDEX           :            dtypes.float                             [10, 14]                         None
+  16 Ops.ADD             :            dtypes.float                             [12, 15]                         None
+  17 Ops.CONST           :            dtypes.weakint                           []                               2
+  18 Ops.CAST            :            dtypes.int                               ['2']                            dtypes.int
+  19 Ops.INDEX           :            dtypes.float                             [10, 18]                         None
+  20 Ops.ADD             :            dtypes.float                             [16, 19]                         None
+  21 Ops.CONST           :            dtypes.weakint                           []                               3
+  22 Ops.CAST            :            dtypes.int                               ['3']                            dtypes.int
+  23 Ops.INDEX           :            dtypes.float                             [10, 22]                         None
+  24 Ops.ADD             :            dtypes.float                             [20, 23]                         None
+  25 Ops.SHRINK          :            dtypes.float                             [6, 8, 8]                        None
+  26 Ops.LOAD            :            dtypes.float                             [25]                             None
+  27 Ops.INDEX           :            dtypes.float                             [26, 2]                          None
+  28 Ops.ADD             :            dtypes.float                             [24, 27]                         None
+  29 Ops.INDEX           :            dtypes.float                             [26, 14]                         None
+  30 Ops.ADD             :            dtypes.float                             [28, 29]                         None
+  31 Ops.INDEX           :            dtypes.float                             [26, 18]                         None
+  32 Ops.ADD             :            dtypes.float                             [30, 31]                         None
+  33 Ops.INDEX           :            dtypes.float                             [26, 22]                         None
+  34 Ops.ADD             :            dtypes.float                             [32, 33]                         None
+  35 Ops.CONST           :            dtypes.weakint                           []                               8
+  36 Ops.CAST            :            dtypes.int                               ['8']                            dtypes.int
+  37 Ops.SHRINK          :            dtypes.float                             [6, 36, 8]                       None
+  38 Ops.LOAD            :            dtypes.float                             [37]                             None
+  39 Ops.INDEX           :            dtypes.float                             [38, 2]                          None
+  40 Ops.ADD             :            dtypes.float                             [34, 39]                         None
+  41 Ops.INDEX           :            dtypes.float                             [38, 14]                         None
+  42 Ops.ADD             :            dtypes.float                             [40, 41]                         None
+  43 Ops.INDEX           :            dtypes.float                             [38, 18]                         None
+  44 Ops.ADD             :            dtypes.float                             [42, 43]                         None
+  45 Ops.INDEX           :            dtypes.float                             [38, 22]                         None
+  46 Ops.ADD             :            dtypes.float                             [44, 45]                         None
+  47 Ops.CONST           :            dtypes.weakint                           []                               12
+  48 Ops.CAST            :            dtypes.int                               ['12']                           dtypes.int
+  49 Ops.SHRINK          :            dtypes.float                             [6, 48, 8]                       None
+  50 Ops.LOAD            :            dtypes.float                             [49]                             None
+  51 Ops.INDEX           :            dtypes.float                             [50, 2]                          None
+  52 Ops.ADD             :            dtypes.float                             [46, 51]                         None
+  53 Ops.INDEX           :            dtypes.float                             [50, 14]                         None
+  54 Ops.ADD             :            dtypes.float                             [52, 53]                         None
+  55 Ops.INDEX           :            dtypes.float                             [50, 18]                         None
+  56 Ops.ADD             :            dtypes.float                             [54, 55]                         None
+  57 Ops.INDEX           :            dtypes.float                             [50, 22]                         None
+  58 Ops.ADD             :            dtypes.float                             [56, 57]                         None
+  59 Ops.SQRT            :            dtypes.float                             [58]                             None
+  60 Ops.NEG             :            dtypes.float                             [59]                             None
+  61 Ops.STORE           :            dtypes.void                              [3, 60]                          None
+  62 Ops.SINK            :            dtypes.void                              [61]                             KernelInfo(name='r_16', applied_opts=(Opt(op=OptOps.SPLIT, axis=0, arg=(0, AxisType.UNROLL)),), opts_to_apply=None, estimates=None, beam=0)
